src/concrete/ml/sklearn/linear_model.py

```diff
@@ -493,102 +556,157 @@ def _fit_encrypted(
 
         # Initialize the weight values with the given ones if some are provided
         if coef_init is not None:
-            weights = coef_init
+            weights = coef_init.reshape(weight_shape)
 
         # Else, if warm start is activated or this is a partial fit, use some already computed
         # weight values have if there are some
         elif (self.warm_start or is_partial_fit) and self._weights_encrypted_fit is not None:
-            weights = self._weights_encrypted_fit
+            weights = self._weights_encrypted_fit.reshape(weight_shape)
 
         # Else, initialize the values randomly
         else:
             weights = self.random_number_generator.uniform(
                 low=self.parameters_range[0],
                 high=self.parameters_range[1],
-                size=(1, X.shape[1], 1),
+                size=weight_shape,
             )
 
         # If the mode should fit the bias values as well
         if self.fit_intercept:
 
             # Initialize the bias values with the given ones if some are provided
             if intercept_init is not None:
-                bias = intercept_init
+                bias = intercept_init.reshape(bias_shape)
 
             # Else, if warm start is activated or this is a partial fit, use some already computed
             # bias values have if there are some
             elif (self.warm_start or is_partial_fit) and self._bias_encrypted_fit is not None:
-                bias = self._bias_encrypted_fit
+                bias = self._bias_encrypted_fit.reshape(bias_shape)
 
             # Else, initialize the values randomly
             else:
                 bias = self.random_number_generator.uniform(
                     low=self.parameters_range[0],
                     high=self.parameters_range[1],
-                    size=(1, 1, 1),
+                    size=bias_shape,
                 )
 
         # Else, initialize the bias with zeros
         else:
-            bias = numpy.zeros((1, 1, 1))
+            bias = numpy.zeros(bias_shape)
 
-        loss_value_moving_average = None
-
-        if self.verbose:
+        # Only print this verbose once if in training using `partial_fit``
+        if self.verbose and (not is_partial_fit or self.training_quantized_module is None):
             mode_string = " (simulation)" if fhe == "simulate" else ""
             print(f"Training on encrypted data{mode_string}...")
 
         # A partial fit is similar to running a fit with a single iteration
         max_iter = 1 if is_partial_fit else self.max_iter
 
-        # Iterate on the training quantized module in the clear
-        for iteration_step in range(max_iter):
+        # Quantize and encrypt the batches
```

Comment:
Not sure about the comment change? Quantization happen further down. The original comment was more on point.

Reply:
yep that's a mistake, good catch !

Reply:
no actually after looking at this, this loop is for quantizing and encrypting the values, the training iteration happens in the following loop